### PR TITLE
Quote discount

### PR DESCRIPTION
**Pull request by ananthisri**

### CODE REVIEW

File: dealer/views.py [hunk 1]
```diff
@@ -1552,12 +1548,14 @@ def quoteView(request):
 
     try:
         quotes = []
+        print("quots::::::::", quote_records)
```
Comment (manti) [inline dealer/views.py]: Use logger not print

File: dealer/views.py [hunk 2]
```diff
@@ -1967,25 +1974,41 @@ def quoteDiscount(request):
...
+                for param in quote_params:
+                    # temp_param = param.replace("-", "_")
+                    temp_param = param
+                    quote_price = float(quote_price) + float(price_dict[temp_param])  
+            request.session['quote_price'] = quote_price
+
+            if outlet_data:
+                max_dealer_discount = DealerDiscount.objects.filter(dealer=outlet_data.dealer, variant=variant, city_id=city_id, is_latest=True).aggregate(test=Coalesce(Max('discount'), 0))
+            else:
+                max_dealer_discount = 0
```
Comment (manti) [inline dealer/views.py]: You can write like this, else is not required
```
max_dealer_discount = 0
if outlet_data:
   max_dealer_discount = DealerDiscount.objects.filter(dealer=outlet_data.dealer, variant=variant, city_id=city_id, is_latest=True).aggregate(test=Coalesce(Max('discount'), 0))
```

File: dealer/views.py [hunk 3]
```diff
@@ -2026,6 +2049,9 @@ def quoteDiscount(request):
                 acko_discount = acko_discount.first()
             else:
                 acko_discount = ""
+
+            if not request.session.get('premium_percent'):
+                request.session['premium_percent'] = 'hundred_percent'
```
Comment (manti) [inline dealer/views.py]: Why do we need this ?

File: dealer/views.py [hunk 4]
```diff
@@ -2112,15 +2138,14 @@ def quotePreview(request, quote_type):
...
-                print("data::::::", event_data)
                 user_data = {
                     "userId": new_quote.phone_number,
                     "phone": new_quote.phone_number,
                     "firstName": new_quote.customer_name,
                     "email":new_quote.email,
                 }
                 engage = quote_webengage(user_data, event_data)
-                print("engage-----", engage.__dict__)
+                # print("engage-----", engage.__dict__)
```
Comment (manti) [inline dealer/views.py]: Use logger, @gokulacko Please help @ananthisri in this

File: dealer/views.py [hunk 5]
```diff
@@ -1967,25 +1974,41 @@ def quoteDiscount(request):
...
+                    for param in non_man:
+                        market_price = float(market_price) + float(price_dict[param])
+            
             request.session['market_price'] = str(market_price)
-            print("market_price::::::::::::::::", market_price)
-            print("quote price::::::::::::::::", request.session.get('quote_price'))
-            print("quote params::::::::::::::::", request.session.get('quote_params'))
 
-            max_dealer_discount = DealerDiscount.objects.filter(variant=variant, city_id=city_id, is_latest=True).aggregate(test=Coalesce(Max('discount'), 0))
+            quote_price = 0.00
```
Comment (manti) [inline dealer/views.py]: 🤔

File: dealer/views.py [hunk 6]
```diff
@@ -1967,25 +1974,41 @@ def quoteDiscount(request):
...
+            quote_params_list = request.session.get('quote_params')
+            if quote_params_list:
+                quote_params = quote_params_list.split(",")
+                price_dict = price_config.__dict__
+                for param in quote_params:
+                    # temp_param = param.replace("-", "_")
+                    temp_param = param
+                    quote_price = float(quote_price) + float(price_dict[temp_param])
```
Comment (manti) [inline dealer/views.py]: float conversation not required in quote_price

File: dealer/views.py [hunk 7]
```diff
@@ -1967,25 +1974,41 @@ def quoteDiscount(request):
             color = request.session.get('color')
 
             outlet = json.loads(request.session['outlet'])
-            print("outlet:::::::", outlet)
+            if outlet['id']:
```
Comment (manti) [inline dealer/views.py]: `if outlet.get('id'):`

File: dealer/views.py [hunk 8]
```diff
@@ -1752,68 +1750,78 @@ def quoteOutlet(request):
...
-        data['offers_count'] = offers_count
-        # print("consumer_offers::::::", consumer_offers.count())
-
-        if dealer_discount:
-            data['dealer_discount'] = dealer_discount.first().discount
-        else:
-            data['dealer_discount'] = 0.00
-
-        market_price = PriceConfig.objects.filter(variant=variant, city_id=city_id, is_latest=True)
+        color = request.session.get('color')
```
Comment (manti) [inline dealer/views.py]: color can be None, color.upper will return `AttributeError: 'NoneType' object has no attribute 'upper'`

File: dealer/views.py (hunk 1 above)
Comment (ananthisri) [reply dealer/views.py]: Noted.

File: dealer/views.py (hunk 3 above)
Comment (ananthisri) [reply dealer/views.py]: Initially, the insurance is assumed as 100% and is set as default. Using this session value, we are displaying the radio option in the html page.

File: dealer/views.py (hunk 8 above)
Comment (ananthisri) [reply dealer/views.py]: The color value is from a dropdown and cannot be 'NoneType'.

**Pull request merged**